tests/qunit/core/common/assets/js/api/core/data.spec.js

```diff
@@ -0,0 +1,439 @@
+			assert.equal( $e.data.getHTTPMethod( 'delete' ), 'DELETE' );
+			assert.equal( $e.data.getHTTPMethod( 'get' ), 'GET' );
+			assert.equal( $e.data.getHTTPMethod( 'update' ), 'PUT' );
+		} );
+
+		QUnit.test( 'getAllowedMethods(): validate all options', ( assert ) => {
+			assert.deepEqual( $e.data.getAllowedMethods( 'create' ), CREATABLE );
+			assert.deepEqual( $e.data.getAllowedMethods( 'delete' ), DELETABLE );
+			assert.deepEqual( $e.data.getAllowedMethods( 'get' ), READABLE );
+			assert.deepEqual( $e.data.getAllowedMethods( 'update' ), EDITABLE );
+		} );
+
+		QUnit.test( 'commandToEndpoint(): command with format', ( assert ) => {
+			const command = 'component/:paramA/command/:paramB',
+				args = { query: {} };
+
+			args.query.paramA = 'valueA';
+			args.query.paramB = 'valueB';
+
+			const endpoint = $e.data.commandToEndpoint( command, args );
+
+			assert.equal( endpoint, 'component/valueA/command/valueB', 'Valid endpoint.' );
+		} );
+
+		QUnit.test( 'commandToEndpoint(): command with query', ( assert ) => {
+			const command = 'component/command',
+				args = { query: {} };
+
+			args.query.paramA = 'valueA';
+			args.query.paramB = 'valueB';
+
+			const endpoint = $e.data.commandToEndpoint( command, args );
+
+			assert.equal( endpoint, 'component/command?paramA=valueA&paramB=valueB&', 'Valid endpoint.' );
+		} );
+
+		QUnit.test( 'commandToEndpoint(): command with query and index', ( assert ) => {
+			const command = 'component/index',
+				args = { query: {} };
+
+			args.query.paramA = 'valueA';
+			args.query.paramB = 'valueB';
+
+			const endpoint = $e.data.commandToEndpoint( command, args );
+
+			assert.equal( endpoint, 'component?paramA=valueA&paramB=valueB&', 'Valid endpoint.' );
+		} );
+
+		QUnit.test( 'endpointToCommand(): endpoints with index and id', ( assert ) => {
+			const Component = class extends ComponentBase {
+					getNamespace() {
+						return 'test-data-commands';
+					}
+
+					defaultData() {
+						return {
+							index: () => {},
+							'test-id': () => {},
+						};
+					}
+				},
+				component = $e.components.register( new Component() );
+
+			const endpointWithIndex = $e.data.endpointToCommand( 'test-data-commands', {} ),
+				endpointWithId = $e.data.endpointToCommand( 'test-data-commands/test-id/some-id', {} );
+
+			assert.equal( endpointWithIndex, component.getNamespace() + '/index' );
+			assert.equal( endpointWithId, component.getNamespace() + '/test-id' );
+		} );
+
+		QUnit.test( 'prepareHeaders(): with GET', ( assert ) => {
+			const requestData = { paramA: 'valueA' },
+				params = $e.data.prepareHeaders( 'get', requestData );
+
+			assert.equal( params.headers[ 'X-WP-Nonce' ], wpApiSettings.nonce );
+		} );
+
+		QUnit.test( 'prepareHeaders(): with POST', ( assert ) => {
+			const requestData = { paramA: 'valueA' },
+				params = $e.data.prepareHeaders( 'create', requestData );
+
+			assert.equal( params.body, '{"paramA":"valueA"}' );
```

Comment:
Does the body prepared by `prepareHeaders`?

Reply:
body is part of the header.